JFRWebSocket.m

```diff
@@ -555,11 +572,15 @@ -(void)processRawMessage:(uint8_t*)buffer length:(NSInteger)bufferLen
             response.bytesLeft = dataLength;
             response.buffer = [NSMutableData dataWithData:data];
         } else {
+            
             if(receivedOpcode == JFROpCodeContinueFrame) {
                 response.bytesLeft = dataLength;
             } else {
+                
                 if([self.delegate respondsToSelector:@selector(websocketDidDisconnect:error:)]) {
-                    [self.delegate websocketDidDisconnect:self error:[self errorWithDetail:@"second and beyond of fragment message must be a continue frame" code:JFRCloseCodeProtocolError]];
+                    dispatch_async(self.queue,^{
+                        [self.delegate websocketDidDisconnect:self error:[self errorWithDetail:@"second and beyond of fragment message must be a continue frame" code:JFRCloseCodeProtocolError]];
```

Comment:
This 8 lines of disconnect logic occurs 8 times – begging to be abstracted into to a method. Could do it in a separate commit though.